tests/approaches/test_interactive_approach.py

```diff
@@ -55,6 +55,21 @@ def test_interactive_learning_approach():
             pass
         # We won't check the policy here because we don't want unit tests to
         # have to train very good models, since that would be slow.
+    # Test interactive_action_strategy random.
+    utils.update_config({
+        "interactive_action_strategy": "random",
+    })
+    interaction_requests = approach.get_interaction_requests()
+    _generate_interaction_results(env.simulate, teacher, train_tasks,
+                                  interaction_requests)
+    # Test that glib falls back to random if no solvable task can be found.
+    utils.update_config({
+        "interactive_action_strategy": "glib",
+        "timeout": 0.0,
```

Comment:
how did this work if you're just catching approachfailure above, not approachtimeout? (see previous comment)

Reply:
what will happen is it times out for every attempted task, leading to a final approach failure